src/c++/perf_analyzer/report_writer.cc

```diff
@@ -235,6 +246,14 @@ ReportWriter::GenerateReport()
           }
         }
       }
+      if (should_output_llm_metrics_) {
+        if (collector_->IsEmpty()) {
```

Comment:
When would this happen?

Reply:
this is just to make sure we have timestamps ready before all the calculations. not really sure when exactly this could happen.

Reply:
It kinda seems like this sort of check should happen earlier in the program flow. Like if the collector is empty, it means that nothing was observed during profiling, and there should be an error or warning over there at that time. And perhaps this shouldn't be an error, maybe it's better UX to just warn and skip outputting the values rather than crashing.

Fine leaving this as is, just food for thought.